### PR TITLE
NoExtraConsecutiveBlankLinesFixer - Add more configuration options related to switch statements

### PR DESCRIPTION
**Pull request by SpacePossum**
following https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2697

### CODE REVIEW

File: tests/Fixer/Whitespace/NoExtraConsecutiveBlankLinesFixerTest.php [hunk 1]
```diff
@@ -952,6 +952,135 @@ public function provideMessyWhitespacesCases()
...
+    
+                            return new ConstantNode($token->getValue());
+                        case 7:
+                            return new ConstantNode($token->getValue());
+                        case 8:
+                            return 8;
+                        default:
+                            echo 1;
+                    }',
+                '<?php
```
Comment (SpacePossum) [inline tests/Fixer/Whitespace/NoExtraConsecutiveBlankLinesFixerTest.php]: **@stof I've tried to recreate the issue you reported (https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2697) here, however it does not fail for me. Can you confirm you have both `throws` and `return` configured on the fixer?**

**Comment by stof** [comment]
@SpacePossum my config is at https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2694#issuecomment-293953049 and I'm using the 2.2.1 release

**Comment by SpacePossum** [comment]
Your configuration for the Fixer resolves to the `@Symfony` set definition;
```
'curly_brace_block',
'extra',
'parenthesis_brace_block',
'square_brace_block',
'throw',
'use'
```
(https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.2/src/RuleSet.php#L82)

Which explains `but line is not removed after return in this case.`, as `return` is not configured.
The reason why `throw` is in as default but `return` is not, is because we asked @ the symfony repo what they would like as default;
https://github.com/symfony/symfony/pull/19167

We never got around making it possible to add the config options specific for switch statements, so I picked the safe option (i.e. do not fix).
I don't see this feature to be added any time soon TBH, but maybe someone wants to follow op on it.

**Comment by stof** [comment]
Well, I said `-1 for return in a switch` in my comment, but I would have added `-1 for throw in a switch` if I had thought about it too. What I was -1 about was the fact of removing lines between cases due to the last statement of the previous case

**Comment by SpacePossum** [comment]
Yeah, looking back on the SF PR things may not have been as clear as they could've been.
For SF project it might be better to configure the fixer to `extra` only in their config file, so cases within `switch` statements are not touched. At least until we've the options to handle `switch` statements in another way.
To sum up the issue; this is a configuration issue and feature request for dealing with `switch` statements in another matter.
This PR is still relevant IMO as it adds new configuration options that I think are good, although these are not the requested options.

**Comment by julienfalque** [comment]
👍

**Comment by keradus** [comment]
:+1: but:
- please rebase
- please say, is it closing #2697 ?

**Comment by SpacePossum** [comment]
* does not close https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2697
* will rebase soon :+1:

File: tests/Fixer/Whitespace/NoExtraConsecutiveBlankLinesFixerTest.php [hunk 2]
```diff
@@ -1054,16 +1050,14 @@ public function provideSwitchCases()
                     default: return 1;}
 
 
-                    // above stays empty'
-                ,
+                    // above stays empty',
```
Comment (SpacePossum) [inline tests/Fixer/Whitespace/NoExtraConsecutiveBlankLinesFixerTest.php]: fixer is bugged :(

**Comment by SpacePossum** [comment]
tests are green

**Comment by SpacePossum** [comment]
with :+1: from @keradus and approve of @julienfalque this on is RTM right?

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request closed without merging**